### PR TITLE
feat: new hook `useAsync`

### PR DESCRIPTION
**Pull request by xobotyi**
## Checklist

- [x] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [x] Does the code have comments in hard-to-understand areas?
- [x] Is there an existing issue for this PR?
  - #33
- [x] Have the files been linted and formatted?
- [x] Have the docs been updated?
- [x] Have the tests been added to cover new hook?
- [x] Have you run the tests locally to confirm they pass?

### CODE REVIEW

File: utility/add-new-hook.js [hunk 1]
```diff
@@ -59,6 +59,16 @@ import { Example } from './example.stories';
 <Canvas>
   <Story story={Example} inline />
 </Canvas>
+
+## Reference
+
+\`\`\`ts
```
Comment (JoeDuncko) [inline utility/add-new-hook.js:65]: What's going on here?

**Review by JoeDuncko — APPROVED** [review]
Besides the `add-new-hook.js` question, LGTM

File: utility/add-new-hook.js (hunk 1 above)
Comment (xobotyi) [reply utility/add-new-hook.js:65]: Just added reference section to hook documentation template

**Pull request merged**